pkg/cmd/apps.go

```diff
@@ -0,0 +1,282 @@
+	}
+
+	var install = &cobra.Command{
+		Use:          "install",
+		Short:        "Install a Kubernetes app",
+		Long:         `Install a Kubernetes app`,
+		Example:      `  k3sup app install [APP]`,
+		SilenceUsage: true,
+	}
+
+	install.Flags().String("kubeconfig", "kubeconfig", "Local path to save the kubeconfig file")
+
+	openfaas := makeInstallOpenFaaS()
+
+	install.RunE = func(command *cobra.Command, args []string) error {
+
+		if len(args) == 0 {
+			fmt.Printf("You can install: %s\n", strings.TrimRight(strings.Join(getApps(), ", "), ", "))
+			return nil
+		}
+
+		return nil
+	}
+
+	command.AddCommand(install)
+	install.AddCommand(openfaas)
+
+	return command
+}
+
+func getApps() []string {
+	return []string{"openfaas"}
+}
+
+func installOpenFaaS(kubeconfigPath string, loadBalancer bool) error {
+
+	res, err := http.Get("https://raw.githubusercontent.com/openfaas/faas-netes/master/install.sh")
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+
+	out, _ := ioutil.ReadAll(res.Body)
+
+	val := string(out)
+	if !loadBalancer {
+		val = strings.Replace(val, "LoadBalancer", "NodePort", -1)
+	}
+
+	script := path.Join(os.TempDir(), "install.sh")
+
+	err = ioutil.WriteFile(script, []byte(val), 0600)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Wrote script to: %s\n", script)
+
+	cmd1 := exec.Command("/bin/bash", script)
+	cmd1.Env = append(os.Environ(), "KUBECONFIG="+kubeconfigPath)
+	cmd1.Stderr = os.Stderr
+	cmd1.Stdout = os.Stdout
+
+	startErr := cmd1.Start()
+	if startErr != nil {
+		return startErr
+	}
+
+	cmd1.Wait()
+
+	return nil
+}
+
+func makeInstallOpenFaaS() *cobra.Command {
+	var openfaas = &cobra.Command{
+		Use:          "openfaas",
+		Short:        "Install openfaas",
+		Long:         `Install openfaas`,
+		Example:      `  k3sup app install openfaas --loadbalancer`,
+		SilenceUsage: true,
+	}
+
+	openfaas.Flags().Bool("loadbalancer", false, "add a loadbalancer")
+	openfaas.Flags().StringP("namespace", "n", "openfaas", "Namepsace for core services")
+
+	openfaas.RunE = func(command *cobra.Command, args []string) error {
+
+		err := config.InitUserDir()
+		if err != nil {
+			return err
+		}
+
+		kubeConfigPath := path.Join(os.Getenv("HOME"), ".kube/config")
+
+		if val, ok := os.LookupEnv("KUBECONFIG"); ok {
+			kubeConfigPath = val
+		}
+		if command.Flags().Changed("kubeconfig") {
+			kubeConfigPath, _ = command.Flags().GetString("kubeconfig")
+		}
+		fmt.Printf("Using context: %s\n", kubeConfigPath)
+
+		lb, _ := command.Flags().GetBool("loadbalancer")
+
+		namespace, _ := command.Flags().GetString("namespace")
+
+		err = addHelmRepo("openfaas", "https://openfaas.github.io/faas-netes/")
+		if err != nil {
+			return err
+		}
+
+		err = updateHelmRepos()
+
+		if err != nil {
+			return err
+		}
+
+		err = installOpenFaaS(kubeConfigPath, lb)
+
+		if err != nil {
+			return err
+		}
+
+		err = kubectl("apply", "-f", "https://raw.githubusercontent.com/openfaas/faas-netes/master/namespaces.yml")
```

Comment:
Is this namespace script needed if you also accept a namespace flag?

Reply:
Good point

Reply:
Ideally we'd template the namespace file or replace text within it.

```
  labels:
    role: openfaas-system
    access: openfaas-system
    istio-injection: enabled
```

^ that or codify these labels

Reply:
.. and call `kubectl create ns`